.github/workflows/set-date.yml

```diff
@@ -0,0 +1,85 @@
+name: Set the End Date in the project
+
+on:
+  issues:
+    types: [closed]
+jobs:
+  set_date:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Get project data
+        env:
+          GITHUB_TOKEN: ${{ secrets.GHA_TOKEN }} 
+          ORGANIZATION: keptn
+          PROJECT_NUMBER: 18
+        run: |
+          gh api graphql -f query='
+            query($org: String!, $number: Int!) {
+              organization(login: $org){
+                projectV2(number: $number) {
+                  id
+                  fields(first:20) {
+                    nodes {
+                      ... on ProjectV2Field {
+                        id
+                        name
+                      }
+                      ... on ProjectV2SingleSelectField {
+                        id
+                        name
+                        options {
+                          id
+                          name
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }' -f org=$ORGANIZATION -F number=$PROJECT_NUMBER > project_data.json
+          echo 'PROJECT_ID='$(jq '.data.organization.projectV2.id' project_data.json) >> $GITHUB_ENV
+          echo 'DATE_FIELD_ID='$(jq '.data.organization.projectV2.fields.nodes[] | select(.name== "End Date") | .id' project_data.json) >> $GITHUB_ENV
+      
+      - name: Get date
+        run: echo "DATE=$(date +"%Y-%m-%d")" >> $GITHUB_ENV
+        
+      - name: Add item id
```

Comment:
In the `Add item id` step, we are extracting the project item ID of the issue and later saving it into an env variable named `ITEM_ID`

Reply:
Why do you need a mutation to get the item ID? 🤔

Reply:
I looked for other ways to get the ID of a project item and I found this [way, ](https://github.com/marketplace/actions/get-project-item-id)but I was facing errors while trying this method. So I went with the mutation method as mentioned [here](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/automating-projects-using-actions#example-workflow-authenticating-with-a-personal-access-token).
Also, I don't know if this is relevant or not but if we go on to use the method which doesn't require mutation, then we would have to provide the `repo` and `org:read` scopes to our token too([as mentioned here](https://github.com/marketplace/actions/get-project-item-id#github-token)). With the mutation method, we just require the `project` scope only.

Reply:
I looked for other ways to get the ID of a project item and I found this [way, ](https://github.com/marketplace/actions/get-project-item-id)but I was facing errors while trying this method. So I went with the mutation method as mentioned [here](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/automating-projects-using-actions#example-workflow-authenticating-with-a-personal-access-token).
Also, I don't know if this is relevant or not but if we go on to use the method which doesn't require mutation, then we would have to provide the `repo` and `org:read` scopes to our token too([as mentioned here](https://github.com/marketplace/actions/get-project-item-id#github-token)). With the mutation method, we just require the `project` scope only.